go-mode.el

```diff
@@ -575,6 +575,18 @@ current line will be returned."
       (skip-chars-forward "^}")
       (forward-char))))
 
+(defun go--add-log-current-defun ()
+  "Return the name of the current defun.
+This is to be used as `add-log-current-defun-function'."
+  ;; TODO this should also deal with types and maybe import.
+  (save-excursion
+    (beginning-of-defun)
```

Comment:
If point is before the `func` keyword, `beginning-of-defun` will go to the previous function. We should probably first check if we're already looking at a function/method. We can anchor the match to the beginning of line, Go functions always start in column 0.